### PR TITLE
Webpack new

### PR DESCRIPTION
**Pull request by StevenWafeek**
In this project we : 
- Add HTML file
- Add Webpack files
- Add TODO list

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
   },
+  module: {
+    rules: [
+      {
+        test: /\.css$/i,
+        use: ['style-loader', 'css-loader'],
+      },
+    ],
+  },
 };
```
Comment (Abenezer-Tilahun) [inline webpack.config.js]: - [ ] Kindly I recommend you to set up the project with Basic commands for building Webpack in **development mode**, or **production mode** is configured and also  **Webpack dev server** is set up and running with no errors.
- [ ] Kindly make sure that has an HTML file configured in the Webpack config settings as a template using the **HTML Webpack Plugin.** , Kindly configure your webpaack just like this : 
```
const path = require('path');
 const HtmlWebpackPlugin = require('html-webpack-plugin');

 module.exports = {
   mode: 'development',
   entry: {
     index: './src/index.js',
     print: './src/print.js',
   },
   devtool: 'inline-source-map',

  devServer: {

    static: './dist',

  },
   plugins: [
     new HtmlWebpackPlugin({
       title: 'Development',
     }),
   ],
   output: {
     filename: '[name].bundle.js',
     path: path.resolve(__dirname, 'dist'),
     clean: true,
   },
module: {
    rules: [
      {
        test: /\.css$/i,
        use: ['style-loader', 'css-loader'],
      },
    ],
  },
  optimization: {

    runtimeChunk: 'single',

  },
 };
```

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "webpack-demo",
+  "name": "webpack",
   "version": "1.0.0",
-  "description": "",
+  "description": "<a name=\"readme-top\"></a>",
   "private": true,
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (Abenezer-Tilahun) [inline package.json:7]: - [ ] Kindly make sure that Webpack dev server is set up and running with no error, So kindly add the following script to easily run the dev server as well:

```
"scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
     "start": "webpack serve --open",
    "build": "webpack"
  },
```

File: src/index.js [hunk 3]
```diff
@@ -1,18 +1,51 @@
...
+    li.innerHTML = ` 
+      <label data-id=${task.id} class="${task.completed ? 'todo-completed' : ''}"> 
+      <input type="checkbox" class="todo-item-check" ${task.completed ? 'checked' : ''}> 
+      ${task.description}    
+      </label>
+      <i class="fas fa-ellipsis-v item-edit-icon"></i> 
+      `;
+    todoList.appendChild(li);
+  });
+};
```
Comment (Abenezer-Tilahun) [inline src/index.js]: - [ ] Kindly make sure that On page load, the task array is rendered dynamically to the DOM. The tasks should appear in order of the index values for each task. but in the case of yours when I run the project it throws an error as shown below and kindly make sure that the task array are rendered dynamically to the DOM.Please Fix the issues.
Capture

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @StevenWafeek     , 

Great job for implemented  the some of the Project Requirement , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻
✔️ No linters error
✔️ You used correct Git Hub-Flow.
✔️ Good README 
✔️ Good Commit Messages 


almost

## Required Changes ♻️
_Check the comments below the review._


## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ABDULALI3468 — APPROVED** [review]
Hi @StevenWafeek , 👍 

# APPROVED 💯 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 
☑️Implemented changes asked by the previous CR.
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification.

> * This is My **GitHub**: GitHub followers
> * This is My **Linkedin**: [Linkedin Badge](https://www.linkedin.com/in/abdul-ali-5400bb216/)&nbsp;
> * This is My **Twitter**: [Twitter Badge](https://twitter.com/mrabdul_ali)&nbsp;

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**